apps/address-service/domains/common/utils/services/providerDetectors.js

```diff
@@ -0,0 +1,57 @@
+const {
+    DadataSearchProvider,
+} = require('@address-service/domains/common/utils/services/search/providers')
+const { DADATA_PROVIDER, GOOGLE_PROVIDER } = require('@address-service/domains/common/constants/providers')
+const {
+    GoogleSuggestionProvider,
+    DadataSuggestionProvider,
+} = require('@address-service/domains/common/utils/services/suggest/providers')
+
+/**
+ * @param {string} geo 'ru'|'en'|'dadata'|'google'|'<some city name>'|...
+ * @returns {AbstractSearchProvider}
+ */
+function getSearchProvider (geo) {
+    /** @type {AbstractSearchProvider} */
+    let searchProvider
+    switch (geo) {
```

Comment:
It's a set of provider names and countries values in variable, that by it's name determines location.

Reply:
It's some value to detect the provider.